inc/passwords/namespace.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+namespace HM\Platform\Security\Passwords;
+
+use function HM\Platform\get_config;
+use WP_Error;
+use ZxcvbnPhp\Zxcvbn;
+
+/**
+ * Bootstrap.
+ */
+function bootstrap() {
+	add_action( 'check_passwords', __NAMESPACE__ . '\\enforce_password_strength', 10, 3 );
+	add_action( 'admin_print_styles-user-edit.php', __NAMESPACE__ . '\\hide_weak_password_prompt' );
+	add_action( 'admin_print_styles-profile.php', __NAMESPACE__ . '\\hide_weak_password_prompt' );
+}
+
+/**
+ * Get minimum required strength for passwords.
+ *
+ * Gets the minimum required "score" for passwords. The score corresponds to
+ * the following UI strings:
+ *
+ * - 1 = "Very Weak"
+ * - 2 = "Weak"
+ * - 3 = "Medium"
+ * - 4 = "Strong"
+ *
+ * @return int Minimum required strength
+ */
+function get_minimum_strength() : int {
+	$config = get_config()['modules']['security'];
+
+	/**
+	 * Filter the minimum required password strength.
+	 *
+	 * Filter the required strength for a password. These scores correspond to
+	 * the following UI strings:
+	 *
+	 * - 1 = "Very Weak"
```

Comment:
Could we maybe actually have these as strings? Why the numbering system if we are just saying it means "weak:/ It would be a lot nicer to be able to do `"minimum-password-strength": "weak"` in the config for example

Reply:
These are just the *current* strings in the UI, but they've already changed. They used to be "bad", "good", "strong" and "strong". The raw score is what we receive from Zxcvbn, so better to stick to that instead for forwards-compatibility if those strings ever change again.

Reply:
Given we pass the numbers to the library, and we are going to document that `2 = weak`, then what's the harm in us taking ownership of the label for that?

Reply:
The Zxcvbn docs use completely different terminology:
```
result.score      # Integer from 0-4 (useful for implementing a strength bar)

  0 # too guessable: risky password. (guesses < 10^3)

  1 # very guessable: protection from throttled online attacks. (guesses < 10^6)

  2 # somewhat guessable: protection from unthrottled online attacks. (guesses < 10^8)

  3 # safely unguessable: moderate protection from offline slow-hash scenario. (guesses < 10^10)

  4 # very unguessable: strong protection from offline slow-hash scenario. (guesses >= 10^10)
```

Also, if the UI strings ever change, it would be way more confusing. If we have `very_weak/weak` but it's now labelled `Unsafe / Very Weak`, you could end up setting the strength incorrectly. By sticking to the exact value and avoiding the abstraction, label updates mean just a doc change rather than potentially confusing behaviour.